### PR TITLE
Summary without chatlist c-object

### LINKED ISSUE #1762: search shows placeholders
on some searches, maybe when typing a bit faster, not sure, only the first few chats are shown, for he others the placehoders stay.

the placeholders are not clickable.

the number of chats in the title is correct.

### PR DESCRIPTION
**Pull request by Simon-Laux**
todo:

- [X] new backend functions 
- [x] modify the virtual list to use the new functions
- [x] use the virtual chatlist also in forward message and profile view (get rid of unneeded code)

related:
requires https://github.com/deltachat/deltachat-node/pull/474
fixes #1762 

note:
doesn't update contact name immediately in chat-list as long as core doesn't throw an chat changed event (see https://github.com/deltachat/deltachat-core-rust/issues/1930 for progress on that bug)

### CODE REVIEW

**Comment by Simon-Laux** [comment]
- [ ] some events don't return a message id, such as chat changed so I need to get the previous message id in this case.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 - Fix automatically login to last remembered account
 - Fix scanning qr code and general improvement of the whole process
 - Fix avatar initials of the text avatars, of people with no displayname, inside of the messagelist
+- Fix search shows placeholders
```
Comment (Jikstra) [inline CHANGELOG.md:22]: What does this mean?

File: src/main/deltachat/chatlist.ts [hunk 2]
```diff
@@ -96,7 +111,67 @@ export default class DCChatList extends SplitOut {
     return chats
   }
 
-  async getChatListSummary(list: ChatList, i: number) {
+  async getChatListItemsByEntries(entries: [number, number][]) {
+    const label = '[BENCH] getChatListItemsByEntries'
```
Comment (Jikstra) [inline src/main/deltachat/chatlist.ts:115]: I think we can uncomment the benchmarking?

**Comment by Jikstra** [comment]
I have the feeling that this could introduce bugs, can we postpone this to the next version/merge window?

File: src/main/deltachat/chatlist.ts (hunk 2 above)
Comment (Simon-Laux) [reply src/main/deltachat/chatlist.ts:115]: yes probably

**Comment by Simon-Laux** [comment]
It's not like I didn't test it.
I know two small bugs, but that is dependent on a new core and then will need like 10 lines of code to fix it:
- Contactname triggered chatname update (only update core, no changes necessary)
- Contactname updates in chat summaries (requires the said lines of code, see https://github.com/deltachat/deltachat-desktop/issues/1889)

File: CHANGELOG.md (hunk 1 above)
Comment (Simon-Laux) [reply CHANGELOG.md:22]: #1762
can you think of a better sentence?

File: src/renderer/components/chat/ChatList.tsx [hunk 3]
```diff
@@ -371,59 +382,129 @@ function useLogic(queryStr: string, showArchivedChats: boolean) {
...
         [chatId]: LoadStatus.LOADED,
       }))
     }
+    if (chatId === 0) {
+      // setChatLoading({})
+      // setChatCache({})
+    } else {
+      if (messageId === 0) {
+        // if no message id is provided it tries to take the old one
+        // workaround to get msgId
```
Comment (Simon-Laux) [inline src/renderer/components/chat/ChatList.tsx:433]: I think I need to redo this workaround as it uses chatListIds which can and probably will be a deprecated snapshot of the value

**Pull request merged**